zkstack_cli/crates/zkstack/src/commands/private_rpc/mod.rs

```diff
@@ -0,0 +1,162 @@
+        msg_private_rpc_docker_image_being_built, msg_private_rpc_initializing_database_for,
+        MSG_CHAIN_NOT_FOUND_ERR, MSG_CHAIN_NOT_INITIALIZED,
+        MSG_PRIVATE_RPC_FAILED_TO_DROP_DATABASE_ERR, MSG_PRIVATE_RPC_FAILED_TO_RUN_DOCKER_ERR,
+    },
+};
+
+#[derive(Subcommand, Debug)]
+pub enum PrivateRpcCommands {
+    /// Initializes private proxy database, builds docker image, runs all migrations and creates docker-compose file
+    Init,
+    /// Run private proxy
+    Run,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct PrivateProxyConfig {
+    pub database_url: Url,
+    pub port: u16,
+    pub create_token_secret: String,
+}
+
+pub(crate) async fn run(shell: &Shell, args: PrivateRpcCommands) -> anyhow::Result<()> {
+    match args {
+        PrivateRpcCommands::Init => init(shell).await,
+        PrivateRpcCommands::Run => run_proxy(shell).await,
+    }
+}
+
+async fn initialize_private_rpc_database(
+    shell: &Shell,
+    chain_config: &ChainConfig,
+    db_url: &Url,
+) -> anyhow::Result<()> {
+    let db_config = db::DatabaseConfig::from_url(db_url)?;
+    db::drop_db_if_exists(&db_config)
+        .await
+        .context(MSG_PRIVATE_RPC_FAILED_TO_DROP_DATABASE_ERR)?;
+    db::init_db(&db_config).await?;
+
+    let db_url = db_config.full_url();
+    let url_str = db_url.as_str();
+    let migrations_dir = chain_config
+        .link_to_code
+        .join("private-rpc")
+        .join("drizzle")
+        .display()
+        .to_string();
+    Cmd::new(cmd!(
+        shell,
+        "cargo sqlx migrate run --database-url {url_str} --source {migrations_dir}"
+    ))
+    .run()?;
+
+    Ok(())
+}
+
+fn prompt_db_config(config: &ChainConfig) -> anyhow::Result<Url> {
+    let chain_name = config.name.clone();
+    logger::info(msg_private_rpc_initializing_database_for(&chain_name));
+    let default_db_name: String = generate_private_rpc_db_name(config);
+
+    let private_rpc_db_url = Prompt::new(&msg_private_rpc_db_url_prompt(&chain_name))
+        .default(DATABASE_PRIVATE_RPC_URL.as_str())
+        .ask();
+    let private_rpc_db_name: String = Prompt::new(&msg_private_proxy_db_name_prompt(&chain_name))
+        .default(&default_db_name)
+        .ask();
+    let private_rpc_db_name = slugify!(&private_rpc_db_name, separator = "_");
+    let db_config = db::DatabaseConfig::new(private_rpc_db_url, private_rpc_db_name);
+
+    Ok(db_config.full_url())
+}
+
+pub async fn init(shell: &Shell) -> anyhow::Result<()> {
+    let ecosystem_config = EcosystemConfig::from_file(shell)?;
+
+    let chain_config = ecosystem_config
+        .load_current_chain()
+        .context(MSG_CHAIN_NOT_INITIALIZED)?;
+    let chain_name = chain_config.name.clone();
+
+    let _dir_guard = shell.push_dir(chain_config.link_to_code.join("private-rpc"));
+
+    logger::info(msg_private_rpc_docker_image_being_built());
+    Cmd::new(cmd!(shell, "docker build -t private-rpc .")).run()?;
```

Comment:
are you sure it's convenient to rebuild docker during the development process? how this process should look like? 

and where is the part with database migrations?

Reply:
It will need to be built exactly once, next times docker will reuse cache, it builds in less than a minute on my machine with cold cache; compared to time needed to compile server it's imo suprisingly convenient. Alternative would be obviously to use already built image, but it would be confusing as we built everything else from source in zksync-era in ZK Stack. The migrations are run during "init" command using sqlx

Reply:
What i'm saying here: 
1. It's a common way how we do it. Probably it'd make sense to have a few alternative way how to run it
2. It's not a common way for js development either.
3. If we need to do migrations during the init and it's the only way how to build new version, would it mean it will erase all data there? if no how we would erase the data?

Reply:
Having a separate DB for this service is very temporary, it will be removed in the coming weeks, so I prefer leaving it how it is. I will build a better UX around the "second iteration" where DB will be only in different service (some sort of admin panel)

Reply:
as for way to build it, it's also likely to change, think of it as a PoC rather than a final version